backend support for updating nav

robdaemon opened the pull request:
making functions to move the human-friendly descriptions of filters and
aggregations and such to the server-side

Commit:
backend support for updating nav

making functions to move the human-friendly descriptions of filters and
aggregations and such to the server-side

Commit:
the basic outline of the descriptor generators is in

Commit:
adding segment support

Commit:
wiring to frontend

Commit:
filling in the description on the forms

Commit:
clean namespace forms

kdoh commented:
Just pulled this down and when I visit the metrics or segments listing pages I get `Don't know how to create ISeq from: clojure.lang.Keyword` in the UI.

Commit:
working on tests, appeasing the linters

Commit:
testing / fixes

no more ISeq error, but has edge cases where query syntaxes are slightly
different. will resolve via the mbql parsing helpers to avoid (first
(second (second ...))) fun.

Commit:
reworked with mbql query helpers

Matches the parts of a query and outputs a data structure that can be
used in the UI

UI wiring is done for display

Commit:
remove unused TableMetadata references
